### PR TITLE
Improve / update Go

### PR DESCRIPTION
**Pull request by Kyslik**
- simplify (shorten) overall text
- simplify example source-code
- reword some sections and update links
- remove GOROOT information, since its not needed anymore (go 1.9+)

### CODE REVIEW

File: Go/README.md [hunk 1]
```diff
@@ -1,74 +1,71 @@
...
 ```sh
-export GOPATH=$HOME/go
-export GOROOT=/usr/local/opt/go/libexec
-export PATH=$PATH:$GOPATH/bin
-export PATH=$PATH:$GOROOT/bin
+export GOPATH=/something-else
 ```
 
-Once you save and close that file, make sure you run `source <filename>` to restart your bash session so that you have those changes. For example:
+Add `GOPATH/bin` directory to your `PATH` environment variable so you can run Go programs anywhere.
```
Comment (Kyslik) [inline Go/README.md]: > you can run Go programs **anywhere**.

Emphasis mine

I think *anywhere* needs to be replaced by something more meaningful.

File: Go/README.md [hunk 2]
```diff
@@ -1,74 +1,71 @@
 # Go
 
-[Go](https://github.com/golang/go) (also known as Golang) is an open source programming language maintained by Google.
+[Go](https://golang.org) is an open source programming language that makes it easy to build simple, reliable, and efficient software.
 
 ## Installation
 
-    $ brew install golang
+You can follow the [official instructions on how to install Go](https://golang.org/doc/install), or you can use homebrew instead:
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
You can follow the [official instructions on how to install Go](https://golang.org/doc/install), or you can use Homebrew instead:
```

File: Go/README.md [hunk 3]
```diff
@@ -1,74 +1,71 @@
...
+
+    brew install go
 
 When installed, try to run `go version` to see the installed version of Go.
```
Comment (hugovk) [inline Go/README.md]: Be positive!
```suggestion
When installed, run `go version` to see the installed version of Go.
```

File: Go/README.md [hunk 4]
```diff
@@ -1,74 +1,71 @@
...
+go version go1.11.4 darwin/amd64
+```
+
 ## Setup your workspace
 
-### Add environment variables
+### The GOPATH and PATH environment variables
 
-Go has a unique approach of managing code where you have a single workspace for all your Go projects. For more information see the [documentation](https://golang.org/doc/code.html#Workspaces).
+The GOPATH environment variable specifies the location of your workspace. It defaults to a directory named `go` inside your home directory (`$HOME/go`).
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
The `GOPATH` environment variable specifies the location of your workspace. It defaults to a directory named `go` inside your home directory (`$HOME/go`).
```

File: Go/README.md [hunk 5]
```diff
@@ -1,74 +1,71 @@
...
 
-First, you'll need to tell Go the location of your workspace. We'll do this by adding some environment variables in your shell config file (usually `.bash_profile`, `.bashrc` or `.zshrc`).
+If you really want to change your GOPATH to something else add GOPATH to your shell/bash/zsh initialization file `.bash_profile`, `.bashrc` or `.zshrc`.
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
If you really want to change your `GOPATH` to something else add `GOPATH` to your shell/bash/zsh initialization file `.bash_profile`, `.bashrc` or `.zshrc`.
```

File: Go/README.md [hunk 6]
```diff
@@ -1,74 +1,71 @@
...
-### Create your workspace
-
-Create the workspace directories tree:
-
-    $ mkdir -p $GOPATH $GOPATH/src $GOPATH/pkg $GOPATH/bin
-
-`$GOPATH/src` This is where your Go projects are located
-`$GOPATH/pkg` A folder that contains every package objects
-`$GOPATH/bin` The compiled binaries home
+Make sure to re-source `source .bash_profile` your current session or simply open new tab within iTerm.
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
Make sure to re-source `source .bash_profile` in your current session or simply open new tab within iTerm.
```

File: Go/README.md [hunk 7]
```diff
@@ -78,47 +75,51 @@ You can then use this package in your Go programs like this:
...
-    $ gofmt -w yourcode.go
+    gofmt -w yourcode.go
+
+You can also format an entire package:
 
-You can also format an entire package (**Note that the command is different from formatting a single file**):
+    go fmt path/to/your/package
 
-    $ go fmt path/to/your/package
+> **Note**: that the command is different from formatting a single file
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
> **Note**: the command is different from formatting a single file.
```

File: Go/README.md [hunk 8]
```diff
@@ -78,47 +75,51 @@ You can then use this package in your Go programs like this:
...
+    godoc -src fmt           # fmt package interface in Go source form
+    godoc -all -http :1234   # documentation for all packages on your machine served through http http://localhost:1234
 
 You need to respect some spec in order to document using `godoc`. More information in the [Godoc documentation](https://blog.golang.org/godoc-documenting-go-code).
 
-## Learn more
+### Learn more
 
-This [interactive tutorial](https://tour.golang.org/) will let you learn more about Go.
+The [interactive tutorial](https://tour.golang.org/) will let you learn more about Go. You can also install it localy by `go get golang.org/x/tour` and running it anywhere by `tour` (given that you added `GOPATH/bin` to your `PATH`).
```
Comment (hugovk) [inline Go/README.md]:
```suggestion
The [interactive tutorial](https://tour.golang.org/) will let you learn more about Go. You can also install it locally by `go get golang.org/x/tour` and running it anywhere by `tour` (given that you added `GOPATH/bin` to your `PATH`).
```

File: Go/README.md (hunk 7 above)
Comment (Kyslik) [reply Go/README.md]: not sure if the trailing `.` - period is needed

File: Go/README.md [hunk 9]
```diff
@@ -78,47 +75,51 @@ You can then use this package in your Go programs like this:
...
 
 To format a single file run:
 
-    $ gofmt -w yourcode.go
+    gofmt -w yourcode.go
+
+You can also format an entire package:
 
-You can also format an entire package (**Note that the command is different from formatting a single file**):
+    go fmt path/to/your/package
```
Comment (simeg) [inline Go/README.md]: Here we say `go fmt` but above we say `gofmt`, I'm guessing this is the correct one?

Comment (simeg) [reply Go/README.md]: Ah didn't see the Note below. Ignore me.

**Comment by simeg** [comment]
LGTM! I'll let @hugovk merge when he's happy with it.

**Comment by hugovk** [comment]
It's good enough, thank you!

**Pull request merged**